src/movies.js

```diff
@@ -8,6 +8,17 @@ function getAllDirectors(movies) {
 
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+function getUniqueList(movies) {
+    let list = [];
+    for (let i = 0; i <movies.length; i++) {
+        let director = movies[i].director;
+        if (list.indexOf(director) < 0) {
+            list.push(director);
+        }
+    }
+    return list;
+}
```

Comment:
This solution works perfectly but the idea was to use .map like we learn in the lectures. I would suggest you try this again using map so you get used to that method. You will be using that a lot moving forward :)